### PR TITLE
Capabilities enabled -> lifecyclestate

### PR DESCRIPTION
**Pull request by zahnno**

### CODE REVIEW

File: src/foam/nanos/crunch/Capability.js [hunk 1]
```diff
@@ -40,7 +41,7 @@ foam.CLASS({
     'name',
     'description',
     'version',
-    'enabled',
+    'status',
```
Comment (Nauna) [inline src/foam/nanos/crunch/Capability.js]: think you mean 'lifecycleState' is a column here

**Comment by Nauna** [comment]
Why do capabilities need LifeCycleState ... I don't understand what a pending or rejected capability would be.

THE CLOSE WAS A MISTAKE _ APOLOGIES

File: src/foam/nanos/crunch/Capability.js [hunk 2]
```diff
@@ -348,7 +346,7 @@ foam.CLASS({
       ],
       documentation: `Checks if a permission or capability string is implied by the current capability`,
       javaCode: `
-        if ( ! this.getEnabled() ) return false;
+        if ( getLifecycleState() == LifecycleState.DELETED || getLifecycleState() == LifecycleState.REJECTED ) return false;
```
Comment (noodlemoodle) [inline src/foam/nanos/crunch/Capability.js:355]: why not getlifecyclestate != active ? does pending also mean enabled?

**Comment by zahnno** [comment]
@Nauna it provides additional object access states, prevents deleting capabilities also enabled isn't used much anymore nor is it supported by our easyDAO. Also there was an intention to deprecate enabled.

File: src/foam/nanos/crunch/Capability.js (hunk 2 above)
Comment (zahnno) [reply src/foam/nanos/crunch/Capability.js:355]: Pending doesn't have much of a use case here - getlifecyclestate != active makes sense.

File: src/foam/nanos/crunch/ServerCrunchService.java [hunk 3]
```diff
@@ -619,7 +620,7 @@ public boolean isRenewable_(X x, String capabilityId, boolean firstCall) {
     Capability capability = (Capability) capabilityDAO.find(capabilityId);
     UserCapabilityJunction ucj = crunchService.getJunction(x, capabilityId);
 
-    if ( ! capability.getEnabled() ) return false;
+    if ( capability.getLifecycleState() == LifecycleState.DELETED || capability.getLifecycleState() == LifecycleState.REJECTED ) return false;
```
Comment (Nauna) [inline src/foam/nanos/crunch/ServerCrunchService.java]: nothing to do with your pr - but makes me wonder why this is here. This should be on the capabilityDAO.

Comment (Nauna) [reply src/foam/nanos/crunch/ServerCrunchService.java]: Ya so looked - in the crunchService we have:
https://github.com/kgrgreer/foam3/blob/28d0cf196a50018c5c22def26327246a689a3f1e/src/foam/nanos/crunch/ServerCrunchService.java#L107
https://github.com/kgrgreer/foam3/blob/28d0cf196a50018c5c22def26327246a689a3f1e/src/foam/nanos/crunch/ServerCrunchService.java#L366
https://github.com/kgrgreer/foam3/blob/28d0cf196a50018c5c22def26327246a689a3f1e/src/foam/nanos/crunch/ServerCrunchService.java#L381
which are all just checking if capability null ... so this here deviates from other call expectations.
We will need to fix this sooner then later...can connect team do this?

File: src/foam/nanos/crunch/Capability.js [hunk 4]
```diff
@@ -326,6 +318,12 @@ foam.CLASS({
       Disables crunchService.isRenewable from propogating anything else in the heirarchy.
       Disables user expiry notifications.
       Disables capability.maybeReOpen()`
+    },
+    {
+      class: 'foam.core.Enum',
+      of: 'foam.nanos.auth.LifecycleState',
+      name: 'lifecycleState',
+      value: 'ACTIVE'
```
Comment (Nauna) [inline src/foam/nanos/crunch/Capability.js]: Can we add documentation here on what we are expecting these states to be. A place to reference our expectations.

perhaps...
**documentation: \`
     Enum value interpretation:
      Pending  - Capability does not return on find or select
      Active     -  Capability does return on find or select
      Rejected - Capability does not return on find or select
      Deleted   - Capability does not return on find or select
   \`**

**Review by Nauna — APPROVED** [review]
made a few comments, but overall this makes sense. So Approve!

File: src/foam/nanos/crunch/Capability.js (hunk 4 above)
Comment (jlhughes) [reply src/foam/nanos/crunch/Capability.js]: Pending/Active work to replace Enabled. 
  Pending - ! enabled.
  Active - enabled.
Rejected - unused. 
Deleted - deleted - but this is a case where we really don't need to 'mark' deleted, but no way around that with Lifecycle.

File: src/foam/nanos/crunch/ServerCrunchService.java (hunk 3 above)
Comment (zahnno) [reply src/foam/nanos/crunch/ServerCrunchService.java]: good catch will update

File: src/foam/nanos/crunch/Capability.js (hunk 4 above)
Comment (zahnno) [reply src/foam/nanos/crunch/Capability.js:326]: okidokes

File: src/foam/nanos/crunch/Capability.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/crunch/Capability.js]: @zahnno yes, believe this should be lifecycleState.  
was your intent to show the current capability status - pending, granted?

**Review by noodlemoodle — APPROVED** [review]

**Pull request merged**